src/sbvr-api/permissions.ts

```diff
@@ -684,6 +684,161 @@ const getAlias = (name: string) => {
 	}
 	return `permissions${permissionsJSON}`;
 };
+
+const rewriteRelationship = memoizeWeak(
+	(
+		value: Relationship,
+		name: string,
+		abstractSqlModel: AbstractSqlModel,
+		permissionsLookup: PermissionLookup,
+		vocabulary: string,
+	) => {
+		let escapedName = sqlNameToODataName(name);
+		if (abstractSqlModel.tables[name]) {
+			escapedName = sqlNameToODataName(abstractSqlModel.tables[name].name);
+		}
+
+		const originalAbstractSQLModel = sbvrUtils.getAbstractSqlModel({
+			vocabulary,
+		});
+
+		const rewrite = (object: AnyObject) => {
+			if (_.has(object, '$') && _.isArray(object['$'])) {
+				const arr = object['$'];
+				if (
+					arr.length === 2 &&
+					_.isArray(arr[1]) &&
+					arr[1].length === 2 &&
+					_.isString(arr[1][0])
+				) {
+					// Skip this if we already shortcut this connection
+					if (arr[1][0].endsWith('$bypass')) {
```

Comment:
Can you destructure/type `arr` please, right now I'm not sure what it actually is and it'd be good to have it documented via the types and var names